VidCoder/Services/ProcessingService.cs

```diff
@@ -1240,15 +1241,223 @@ public void Process(string source, string destination, string presetName, string
+						this.QueueSingleSourcePath(array_timespan[i], source, new_destination, presetName, profile, picker);
+					}
+				}
+				else
+				{
+					MessageBox.Show("Skipped File - invalid timespan");
+				}
+			}
+			else
+			{
+				// found single timespan
+				bool result_rangecheck = this.CLI_CheckTimespanInRange(cli_timespan, source, destination, presetName, profile, picker);
+				if (result_rangecheck == false) { rangecheck_errors += 1; }
+				if (rangecheck_errors == 0)
+				{
+					// Convert only when the timespan is in File Range
+					this.QueueSingleSourcePath(cli_timespan, source, destination, presetName, profile, picker);
+				}
+				else
+				{
+					MessageBox.Show("Skipped File - invalid timespan");
+				}
+
+			}
+		}
+
+		public bool CLI_CheckTimespanInRange(string cli_timespan, string source, string destination, string presetName, VCProfile profile = null, Picker picker = null)
+		{
+			var scanMultipleDialog = new ScanMultipleDialogViewModel(new List<SourcePath> { new SourcePath { Path = source } });
+			this.windowManager.OpenDialog(scanMultipleDialog);
+
+			VideoSource videoSource = scanMultipleDialog.ScanResults[0].VideoSource;
+
+			List<int> titleNumbers;
+			if (videoSource != null)
+			{
+				titleNumbers = this.PickTitles(videoSource, picker);
+			}
+			else
+			{
+				titleNumbers = new List<int>();
+			}
+
+			foreach (int titleNumber in titleNumbers)
+			{
+				var jobVM = new EncodeJobViewModel(new VCJob
+				{
+					SourcePath = source,
+					SourceType = Utilities.GetSourceType(source),
+					Title = titleNumber,
+					RangeType = VideoRangeType.All,
+					EncodingProfile = profile,
+					AudioTracks = new List<ChosenAudioTrack> { new ChosenAudioTrack { TrackNumber = 1 } },
+					FinalOutputPath = destination,
+					UseDefaultChapterNames = true,
+					PassThroughMetadata = picker.PassThroughMetadata
+				});
+
+				jobVM.VideoSource = videoSource;
+				jobVM.PresetName = presetName;
+				jobVM.ManualOutputPath = !string.IsNullOrWhiteSpace(destination);
+
+				VCJob job = jobVM.Job;
+
+				SourceTitle title = jobVM.VideoSource.Titles.Single(t => t.Index == titleNumber);
+				jobVM.Job.Length = title.Duration.ToSpan();
+
+				try
+				{
+					if (cli_timespan != null) // timespan CLI Parameter
+					{
+						string[] times = cli_timespan.Split('-');
+						job.RangeType = VideoRangeType.Seconds;
+						TimeSpan timespanStart = TimeSpan.Parse(times[0]);
+						TimeSpan timespanEnd = TimeSpan.Parse(times[1]);
+
+						if (timespanStart >= timespanEnd | timespanEnd >= title.Duration.ToSpan())
+						{
+							return false; // Timespan out of Range
+						}
+						else
+						{
+							return true; // Timespan is in Range
+						}
+					}
+				}
+				catch
+				{
+					return false;
+				}
+
+				return true;
+			}
+			return false;
+		}
+
+		public void QueueSingleSourcePath(string cli_timespan, string source, string destination, string presetName, VCProfile profile = null, Picker picker = null)

```

Comment:
This looks similar to another method I already have, and is sharing code with CLI_CheckTimespanInRange. I think what you want to shoot for is re-using the existing methods but adding an optional range parameter to them. We don't want to copy/paste/modify large methods.